tools/jit/gen_jit_dispatch.py

```diff
@@ -149,18 +151,40 @@ def from_ivalue(arg, value):
 );
 """)
 CALL_NAMESPACE_WITH_TENSOR_OPTIONS = CodeTemplate("""\
+// This is a hack.
+// Please see [Add requires_grad to native_functions.yaml and potentially to TensorOptions object]
+// In the tracking issue: https://github.com/pytorch/pytorch/issues/30405
+const auto options = TensorOptions()
+        .dtype(${dtype})
+        .layout(${layout})
+        .device(${device})
+        .pinned_memory(${pin_memory});
+#ifdef USE_STATIC_DISPATCH
+    auto result_ = at::${name}(${args_with_tensor_options_disp});
+#else
+    auto result_ = torch::_${name}(${args_with_tensor_options});
+#endif
+""")
+CALL_NAMESPACE_WITH_TENSOR_OPTIONS_ARANGE = CodeTemplate("""\
+// This is a hack.
```

Comment:
This is called out as a hack. That's a good first step. What is the hack doing? What is the intended resolution of the hack? In the quoted issue you say "Add requires_grad to native_functions.yaml and potentially to TensorOptions object
Ideally, requires_grad has to be added to all function in native_functions.yaml that have TensorOptions in it. Its not the case right now so there are many places in code gen that add requires_grad on a fly which adds significant complexity to overall logic." Fine, but that doesn't tell me what's going on with this arange special case. What makes arange special?